modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaVertXWebServerCodegen.java

```diff
@@ -0,0 +1,162 @@
+
+    public String getHelp() {
+        return "Generates a Java Vert.x-Web Server.";
+    }
+
+    @Override
+    public void processOpts() {
+        super.processOpts();
+
+        apiTestTemplateFiles.clear();
+
+        importMapping.remove("JsonCreator");
+        importMapping.remove("com.fasterxml.jackson.annotation.JsonProperty");
+        importMapping.put("JsonInclude", "com.fasterxml.jackson.annotation.JsonInclude");
+        importMapping.put("JsonProperty", "com.fasterxml.jackson.annotation.JsonProperty");
+        importMapping.put("JsonValue", "com.fasterxml.jackson.annotation.JsonValue");
+        importMapping.put("FileUpload", "io.vertx.ext.web.FileUpload");
+
+        modelDocTemplateFiles.clear();
+        apiDocTemplateFiles.clear();
+
+        String sourcePackageFolder = sourceFolder + File.separator + invokerPackage.replace(".", File.separator);
+        supportingFiles.clear();
+        supportingFiles.add(new SupportingFile("supportFiles/openapi.mustache", resourceFolder, "openapi.yaml"));
+        supportingFiles.add(new SupportingFile("supportFiles/HttpServerVerticle.mustache", sourcePackageFolder, "HttpServerVerticle.java"));
+        supportingFiles.add(new SupportingFile("supportFiles/MainVerticle.mustache", sourcePackageFolder, "MainVerticle.java"));
+        supportingFiles.add(new SupportingFile("supportFiles/ApiResponse.mustache", sourcePackageFolder, "ApiResponse.java"));
+        supportingFiles.add(new SupportingFile("supportFiles/ApiException.mustache", sourcePackageFolder, "ApiException.java"));
+        supportingFiles.add(new SupportingFile("supportFiles/ParameterCast.mustache", sourcePackageFolder, "ParameterCast.java"));
+        supportingFiles.add(new SupportingFile("supportFiles/pom.mustache", "", "pom.xml"));
+
+        writeOptional(outputFolder, new SupportingFile("README.mustache", "", "README.md"));
+    }
+
+    @Override
+    public void postProcessModelProperty(CodegenModel model, CodegenProperty property) {
+        super.postProcessModelProperty(model, property);
+        if (!model.isEnum) {
+            model.imports.add("JsonInclude");
+            model.imports.add("JsonProperty");
+            if (model.hasEnums) {
+                model.imports.add("JsonValue");
+            }
+        }
+    }
+
+    @Override
+    public Map<String, Object> postProcessOperationsWithModels(Map<String, Object> objs, List<Object> allModels) {
+        Map<String, Object> newObjs = super.postProcessOperationsWithModels(objs, allModels);
+        Map<String, Object> operations = (Map<String, Object>) newObjs.get("operations");
+        if (operations != null) {
+            List<CodegenOperation> ops = (List<CodegenOperation>) operations.get("operation");
+            for (CodegenOperation operation : ops) {
+                operation.httpMethod = operation.httpMethod.toLowerCase(Locale.ROOT);
+
+                if (operation.returnType == null) {
+                    operation.returnType = "Void";
```

Comment:
Suggestion: you may want to use {{^returnType}} ... {{/returnType}} to identify void instead.

If you assign "Void" to returnType, there will be no easy way to detect void return in the mustache template.

Reply:
I was using `{{^returnType}}` in the beginning, but soon realised forcing all null return to Void makes the template file cleaner since all API will be wrapped in RxJava `Single<ApiResponse<{{returnType}}>>`

Reply:
Up to you. (later you may have a use case in which you need to determine whether the operation is returning anything and we can always find a solution/workaround later as this is not a day 1 requirement)